frontend/src/component/player-form/index.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+export default class PlayerForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      firstName: props.player ? props.player.firstName : '',
+      lastName: props.player ? props.player.lastName : '',
+      role: 'player',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    const { name, value } = e.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    let player = { ...this.state };
+
+    if (this.props.player) {
+      player = { ...this.state, _id: this.props.player._id };
```

Comment:
Since you already object spread this.state into player a few lines above you could just make this line as follows:

player = { ...player, _id: this.props.player._id };

Or make your player variable declaration above just an empty object. But spreading this.state twice back to back seems unnecessary.